### ISSUE TITLE
xref crashes with AttributeError: 'LateAnnotation' object has no attribute 'Bindings'

### ISSUE THREAD

**Issue by kamahen** [google/pytype#269]
Clone https://github.com/kamahen/pykythe.git
I don't think you need to install other the other pre-reqs to regenerate this bug; just run:
```
make pytype
```
You should see these lines in the output:
```
pykythe/ast_cooked.py pytype_output/imports/pykythe.ast_cooked.imports /tmp/ast_cooked.xref.json
Pytype error: AttributeError: 'LateAnnotation' object has no attribute 'Bindings'
```

**Comment by rchen152** [comment]
@martindemello would be the best person to look at this one.

**Comment by martindemello** [comment]
thanks, i'll take a look. @kamahen, could you send a PR for the changes you made to get xref to run as a binary?

**Comment by kamahen** [comment]
@martindemello -- I decided to rename `xref` to `pyxref`. Here's how I installed it. (No doubt there's a way to run `xref` without installing it, but I couldn't figure out an appropriate incantation for using `parser_ext.so`):
```diff
$ git diff
diff --git a/pytype/tools/xref/indexer.py b/pytype/tools/xref/indexer.py
index 3fa3e191..cadb8f27 100644
--- a/pytype/tools/xref/indexer.py
+++ b/pytype/tools/xref/indexer.py
@@ -16,7 +16,7 @@ from pytype import module_utils
 from pytype.tools.xref import utils as xref_utils
 from pytype.tools.xref import kythe
 
-from typed_ast import ast27t as ast27
+from typed_ast import ast27 as ast27
 from typed_ast import ast3
 
 
diff --git a/setup.py b/setup.py
index 70054d54..50630f6b 100644
--- a/setup.py
+++ b/setup.py
@@ -126,6 +126,7 @@ setup(
         'pytype/pytd',
         'pytype/pytd/parse',
         'pytype/tools',
+        'pytype/tools/xref',
         'pytype/tools/analyze_project',
         'pytype/tools/merge_pyi',
         'pytype/typegraph',
@@ -136,6 +137,7 @@ setup(
             'pytype-single = pytype.main:main',
             'pytd = pytype.pytd.main:main',
             'pytype = pytype.tools.analyze_project.main:main',
+            'pyxref = pytype.tools.xref.main:main',
         ]
     },
     package_data={'pytype': get_builtin_files()},
```

**Comment by kamahen** [comment]
The "install" command was (after `cd`-ing into the `pytype` source directory):
```bash
pip3 install --upgrade --user .
```
or
```bash
python3.6 -m pip install --upgrade --user .
```

**Comment by kamahen** [comment]
I've downloaded the latest version of `pytype` and turned on the `--debug` flag. This is what I see in the traceback (the error from `/opt/kythe/tools/write_entries` can be ignored; it seems to be due to no JSON output when `pyxref` crashes ... without the `--debug` flag, I see `Run with --debug to see a traceback.`):
```
$ time make pytype
rm -f -r .pytype
time pytype -k --exclude=pykythe/bootstrap_builtins.py pykythe
Computing dependencies
Analyzing 7 sources with 0 local dependencies
ninja: Entering directory `/home/peter/src/pykythe/.pytype'
[1/7] check pykythe.__init__
[2/7] check pykythe.typing_debug
[3/7] check pykythe.pod
[4/7] check pykythe.ast
[5/7] check pykythe.ast_cooked
[6/7] check pykythe.ast_raw
[7/7] check pykythe.__main__
Success: no errors found

real	0m32.226s
user	0m32.884s
sys	0m0.761s
for i in $(ls pykythe/*.py | fgrep -v bootstrap_builtins.py); do \
    echo $i \
        .pytype/imports/pykythe.$(basename $i .py).imports \
	/tmp/$(basename $i .py).pyxref.json; \
    time pyxref --debug --python_version=3.6 --imports_info=.pytype/imports/pykythe.$(basename $i .py).imports $i >/tmp/$(basename $i .py).pyxref.json;  \
done
pykythe/ast_cooked.py .pytype/imports/pykythe.ast_cooked.imports /tmp/ast_cooked.pyxref.json
Pytype error: AttributeError: 'LateAnnotation' object has no attribute 'Bindings'
Traceback (most recent call last):
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 293, in wrap_pytype_exceptions
    yield
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/tools/xref/indexer.py", line 1355, in process_file
    tracer_vm=vm)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 667, in infer_types
    tracer.exitpoint = tracer.analyze(loc, defs, maximum_depth)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 405, in analyze
    return self.analyze_toplevel(node, defs)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 371, in analyze_toplevel
    new_node = self.analyze_class(node, value)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 353, in analyze_class
    node = self.analyze_method_var(node, name, b)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 203, in analyze_method_var
    node2 = self.maybe_analyze_method(node1, val)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 157, in maybe_analyze_method
    node, _ = self.call_function_with_args(node, val, args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 121, in call_function_with_args
    new_node, ret = self.call_function_in_frame(node, fvar, *args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 132, in call_function_in_frame
    state, var, args, kwargs, starargs, starstarargs)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 884, in call_function_with_state
    starstarargs=starstarargs), fallback_to_unsolvable, allow_noreturn=True)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 920, in call_function
    new_node, one_result = func.call(node, funcv, args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 3133, in call
    alias_map=self.alias_map)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 2940, in call
    node2, ret = self.vm.run_frame(frame, node)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 275, in run_frame
    state = self.run_instruction(op, state)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 236, in run_instruction
    state = bytecode_fn(state, op)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1835, in byte_LOAD_ATTR
    state, val = self.load_attr(state, obj, name)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1144, in load_attr
    node, result, errors = self._retrieve_attr(state.node, obj, attr)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1115, in _retrieve_attr
    node, val.data, attr, val)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 66, in get_attribute
    return self._get_instance_attribute(node, obj, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 235, in _get_instance_attribute
    return self._get_attribute(node, obj, obj.cls, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 271, in _get_attribute
    node, attr = self.get_attribute(node, cls, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 59, in get_attribute
    return self._get_class_attribute(node, obj, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 230, in _get_class_attribute
    return self._get_attribute(node, cls, meta, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 285, in _get_attribute
    attr = self._filter_var(node, attr)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 427, in _filter_var
    bindings = var.Bindings(node)
AttributeError: 'LateAnnotation' object has no attribute 'Bindings'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/tools/xref/main.py", line 33, in main
    ix = indexer.process_file(options, kythe_args=kythe_args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/tools/xref/indexer.py", line 1355, in process_file
    tracer_vm=vm)
  File "/usr/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 314, in wrap_pytype_exceptions
    raise exception_type(msg).with_traceback(e.__traceback__)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 293, in wrap_pytype_exceptions
    yield
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/tools/xref/indexer.py", line 1355, in process_file
    tracer_vm=vm)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 667, in infer_types
    tracer.exitpoint = tracer.analyze(loc, defs, maximum_depth)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 405, in analyze
    return self.analyze_toplevel(node, defs)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 371, in analyze_toplevel
    new_node = self.analyze_class(node, value)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 353, in analyze_class
    node = self.analyze_method_var(node, name, b)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 203, in analyze_method_var
    node2 = self.maybe_analyze_method(node1, val)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 157, in maybe_analyze_method
    node, _ = self.call_function_with_args(node, val, args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 121, in call_function_with_args
    new_node, ret = self.call_function_in_frame(node, fvar, *args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 132, in call_function_in_frame
    state, var, args, kwargs, starargs, starstarargs)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 884, in call_function_with_state
    starstarargs=starstarargs), fallback_to_unsolvable, allow_noreturn=True)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 920, in call_function
    new_node, one_result = func.call(node, funcv, args)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 3133, in call
    alias_map=self.alias_map)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 2940, in call
    node2, ret = self.vm.run_frame(frame, node)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 275, in run_frame
    state = self.run_instruction(op, state)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 236, in run_instruction
    state = bytecode_fn(state, op)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1835, in byte_LOAD_ATTR
    state, val = self.load_attr(state, obj, name)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1144, in load_attr
    node, result, errors = self._retrieve_attr(state.node, obj, attr)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1115, in _retrieve_attr
    node, val.data, attr, val)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 66, in get_attribute
    return self._get_instance_attribute(node, obj, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 235, in _get_instance_attribute
    return self._get_attribute(node, obj, obj.cls, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 271, in _get_attribute
    node, attr = self.get_attribute(node, cls, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 59, in get_attribute
    return self._get_class_attribute(node, obj, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 230, in _get_class_attribute
    return self._get_attribute(node, cls, meta, name, valself)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 285, in _get_attribute
    attr = self._filter_var(node, attr)
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 427, in _filter_var
    bindings = var.Bindings(node)
pytype.tools.xref.indexer.PytypeError: Pytype error: AttributeError: 'LateAnnotation' object has no attribute 'Bindings'

real	0m8.992s
user	0m8.832s
sys	0m0.157s
pykythe/ast.py .pytype/imports/pykythe.ast.imports /tmp/ast.pyxref.json

real	0m1.117s
user	0m1.093s
sys	0m0.025s
pykythe/ast_raw.py .pytype/imports/pykythe.ast_raw.imports /tmp/ast_raw.pyxref.json

real	0m21.964s
user	0m21.394s
sys	0m0.568s
pykythe/__init__.py .pytype/imports/pykythe.__init__.imports /tmp/__init__.pyxref.json

real	0m0.844s
user	0m0.803s
sys	0m0.037s
pykythe/__main__.py .pytype/imports/pykythe.__main__.imports /tmp/__main__.pyxref.json

real	0m1.605s
user	0m1.583s
sys	0m0.022s
pykythe/pod.py .pytype/imports/pykythe.pod.imports /tmp/pod.pyxref.json

real	0m1.243s
user	0m1.203s
sys	0m0.041s
pykythe/typing_debug.py .pytype/imports/pykythe.typing_debug.imports /tmp/typing_debug.pyxref.json

real	0m0.742s
user	0m0.695s
sys	0m0.048s
rm -f -r /tmp/pytype-graphstore /tmp/pytype-tables
mkdir -p tmp/pytype-graphstore /tmp/pytype-tables
set -o pipefail; \
    cat /tmp/*.pyxref.json | \
    /opt/kythe/tools/entrystream --read_format=json | \
    /opt/kythe/tools/write_entries -graphstore /tmp/pytype-graphstore
2019/04/03 11:24:37 error decoding JSON Entry: json: cannot unmarshal number into Go value of type storage_go_proto.Entry
write_entries: 2019/04/03 11:24:37 Wrote 0 entries
Makefile:506: recipe for target 'pytype' failed
make: *** [pytype] Error 1

real	1m8.886s
user	1m8.578s
sys	0m1.731s
```

**Comment by martindemello** [comment]
I spent some time poking at this within attribute.py but haven't gotten anywhere yet. If you have some time to help with this, my planned next step is to try stripping down `ast_cooked.py` to isolate a minimal repro case.

**Comment by kamahen** [comment]
Do you have any guesses as to the most likely cause? Presumably it's a problem with the "." operator? (I have only rudimentary knowledge of how `pytype` works.)
The code uses dataclasses wrappers, but so do other source files that don't cause a crash.

**Comment by martindemello** [comment]
no, it looks like it's coming from pytype rather than xref, but running pytype directly on the file doesn't crash so i'm not sure if we're not doing some setup correctly in xref. the proximate cause seems to be a late annotation that doesn't get resolved to a class before being used as one.

**Comment by kamahen** [comment]
Is there a "--noquick" option for pytype? I'm curious how the detailed
analysis differs from the quick analysis (and that might also hold clues to
this bug). And/or is there some kind of "trace" option to pytype, so that I
can see what order it looks at things and maybe notice something that's
unusual?

**Comment by kamahen** [comment]
I got pytype to crash; it swallows the error (and sets RC to 0) but writes the traceback to the `.pyi` file (this should probably be filed as a separate bug):
```
~/src/pykythe$ rm -r .pytype ; time pytype pykythe/ast_cooked.py 
rm: cannot remove '.pytype': No such file or directory
Computing dependencies
Analyzing 1 sources with 3 local dependencies
ninja: Entering directory `/home/peter/src/pykythe/.pytype'
[1/4] infer pykythe.typing_debug
[2/4] infer pykythe.pod
[3/4] infer pykythe.ast
[4/4] check pykythe.ast_cooked
Success: no errors found

real	0m11.290s
user	0m11.622s
sys	0m0.340s

~/src/pykythe$ cat .pytype/pyi/pykythe/ast_cooked.pyi 

from typing import Any
def __getattr__(name: Any) -> Any: ...
# Caught error in pytype: 'LateAnnotation' object has no attribute 'Bindings'
# Traceback (most recent call last):
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 132, in check_or_generate_pyi
#     loader=loader)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 82, in generate_pyi
#     analyze.infer_types, input_filename, errorlog, options, loader)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 57, in _call
#     deep=deep)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 667, in infer_types
#     tracer.exitpoint = tracer.analyze(loc, defs, maximum_depth)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 405, in analyze
#     return self.analyze_toplevel(node, defs)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 371, in analyze_toplevel
#     new_node = self.analyze_class(node, value)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 353, in analyze_class
#     node = self.analyze_method_var(node, name, b)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 203, in analyze_method_var
#     node2 = self.maybe_analyze_method(node1, val)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 157, in maybe_analyze_method
#     node, _ = self.call_function_with_args(node, val, args)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 121, in call_function_with_args
#     new_node, ret = self.call_function_in_frame(node, fvar, *args)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/analyze.py", line 132, in call_function_in_frame
#     state, var, args, kwargs, starargs, starstarargs)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 884, in call_function_with_state
#     starstarargs=starstarargs), fallback_to_unsolvable, allow_noreturn=True)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 920, in call_function
#     new_node, one_result = func.call(node, funcv, args)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 3133, in call
#     alias_map=self.alias_map)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/abstract.py", line 2940, in call
#     node2, ret = self.vm.run_frame(frame, node)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 275, in run_frame
#     state = self.run_instruction(op, state)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 236, in run_instruction
#     state = bytecode_fn(state, op)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1835, in byte_LOAD_ATTR
#     state, val = self.load_attr(state, obj, name)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1144, in load_attr
#     node, result, errors = self._retrieve_attr(state.node, obj, attr)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/vm.py", line 1115, in _retrieve_attr
#     node, val.data, attr, val)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 66, in get_attribute
#     return self._get_instance_attribute(node, obj, name, valself)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 235, in _get_instance_attribute
#     return self._get_attribute(node, obj, obj.cls, name, valself)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 271, in _get_attribute
#     node, attr = self.get_attribute(node, cls, name, valself)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 59, in get_attribute
#     return self._get_class_attribute(node, obj, name, valself)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 230, in _get_class_attribute
#     return self._get_attribute(node, cls, meta, name, valself)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 285, in _get_attribute
#     attr = self._filter_var(node, attr)
#   File "/home/peter/.local/lib/python3.6/site-packages/pytype/attribute.py", line 427, in _filter_var
#     bindings = var.Bindings(node)
# AttributeError: 'LateAnnotation' object has no attribute 'Bindings'
```

**Comment by martindemello** [comment]
running pytype with `-v 4` will trace through its execution, but it's pretty noisy, which is why i wanted to strip down the file first.

**Comment by kamahen** [comment]
What is a "LateAnnotation"? Is it a situation like this?
```python
def foo():
   x = SomeClass()
   ...

class SomeClass:
   ...
```

Or a type annotation like this?:
```python
def foo(x: 'SomeClass'):
   ...

class SomeClass:
   ...
```

**Comment by kamahen** [comment]
> running pytype with `-v 4` will trace through its execution

`-v 4` or `-v 3` gets an error (`CRITICAL:root:Bad verbosity level: 4`); `-v 2` doesn't produce much information. Is there a trick to getting more verbosity?

**Comment by martindemello** [comment]
hm - that might be a bug with our arg handling. i'll look into it.

**Comment by rchen152** [comment]
Peter, are you running `pytype` or `pytype-single`? The pytype executable only has verbosity levels 0 through 2, whereas pytype-single has up to verbosity 4.

**Comment by kamahen** [comment]
Rebecca: I was running `pytype`, not `pytype-single`. So, I tried `pytype-single`, but it appears to need different options than `pytype` (it didn't produce any `.pyi` file and it didn't crash).

**Comment by rchen152** [comment]
They're different tools, yes: https://github.com/google/pytype#subtools. I mentioned `pytype-single` because I assumed that was the one Martin was referring to in his comment about tracing through the execution.

**Comment by kamahen** [comment]
I ran `pytype -v 2`, which produced detailed `pytype-single` commands, and then ran the appropriate `pytype-single` command with `-v 4`. I now have a line number of where things broke, and I'll see if I can make a small repro case.

**Comment by kamahen** [comment]
Here are the files with a smaller repo in `ast_cooked_pytype.py` (plus some extra stuff because I was too lazy to remove them): [ppp.zip](https://github.com/google/pytype/files/3041149/ppp.zip)

Run with this command:
```
cd pykythe
pytype-single -v 4 --imports_info .pytype/imports/pykythe.ast_cooked_pytype.imports --module-name pykythe.ast_cooked_pytype -V 3.6 -o .pytype/pyi/pykythe/ast_cooked_pytype.pyi --analyze-annotated --nofail --quick pykythe/ast_cooked_pytype.py 2>/tmp/pp2.out
```

The problem is triggered by line 60 of `pykythe/ast_cooked_pytype.py`:
```python
        class_fqn = ctx.fqn_dot + self.name.name.value
```

You'll see this near the end of `/tmp/pp2.out`:
```
INFO:pytype.vm: >  60: 2: LOAD_FAST self
INFO:pytype.vm: >  | index: 3, 'add_fqns', module: pykythe.ast_cooked_pytype line: 60
INFO:pytype.vm: >  | data_stack: (<Variable v2501: 1 choices>, <Variable v2508: 1 choices>)
INFO:pytype.vm: >  | block_stack: ()
INFO:pytype.vm: >  | node: <154>add_fqns
INFO:pytype.vm: >  60: 3: LOAD_ATTR name
DEBUG:pytype.vm:LOAD_ATTR: <Variable v2508: 1 choices> 'name'
DEBUG:pytype.vm:getting attr name from <Variable v2508: 1 choices>
WARNING:pytype.io:***Caught exception: 'LateAnnotation' object has no attribute 'Bindings'
Traceback (most recent call last):
  File "/home/peter/.local/lib/python3.6/site-packages/pytype/io.py", line 132, in check_or_generate_pyi
    loader=loader)
```

**Comment by kamahen** [comment]
If this isn't a small enough test file, I can probably make a significantly smaller one (combining some of the imported files and stripping out unused stuff).

**Comment by kamahen** [comment]
Seems to have been fixed, possibly with Google sync #430.

Leaving this open in case there are other related issues (I wasn't the only person on this Issue).

**Comment by rchen152** [comment]
Thanks for checking! I'll go ahead and close this issue, since the other report of a similar crash already has its own bug.
